services/src/datasets/postgres.rs

```diff
@@ -0,0 +1,171 @@
+use crate::datasets::listing::{DataSetListOptions, DataSetListing, DataSetProvider};
+use crate::datasets::storage::{
+    AddDataSet, AddDataSetProvider, DataSetDB, DataSetPermission, DataSetProviderListOptions,
+    DataSetProviderListing, ImportDataSet, RasterLoadingInfo, VectorLoadingInfo,
+};
+use crate::error::Result;
+use crate::users::user::UserId;
+use crate::util::user_input::Validated;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use geoengine_datatypes::collections::{FeatureCollection, TypedFeatureCollection};
+use geoengine_datatypes::dataset::{
+    DataSetId, DataSetProviderId, InternalDataSetId, StagingDataSetId,
+};
+use geoengine_datatypes::primitives::Geometry;
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use geoengine_operators::engine::{
+    LoadingInfo, LoadingInfoProvider, RasterResultDescriptor, VectorResultDescriptor,
+};
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+
+#[derive(Debug)]
+pub struct PostgresDataSetDB {}
+
+#[async_trait]
+impl DataSetDB for PostgresDataSetDB {
```

Comment:
This is totally todo, is there a reason to have it implemented?

Reply:
should be a next step then

Reply:
I need to have it because the `Context` requires a DataSetDB and there is a PostgresContext so I need a PostgresDataSetDB